kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/InsecureTls.java

```diff
@@ -6,20 +6,19 @@
 
 package io.kroxylicious.proxy.config.tls;
 
-import io.netty.handler.ssl.SslContextBuilder;
-import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import com.fasterxml.jackson.annotation.JsonInclude;
 
 /**
  * A {@link TrustProvider} that allows the disabling trust verification.
  * <strong>Not recommended for production use.</strong>
  *
  * @param insecure true if trust is disabled.  If false, platform trust is used.
  */
-public record InsecureTls(boolean insecure) implements TrustProvider {
+public record InsecureTls(@JsonInclude(JsonInclude.Include.ALWAYS) boolean insecure) implements TrustProvider {
```

Comment:
why include always?

Reply:
I tried writing an integration test using `InsecureTls(false)` this does not survive the conversion we do because the ConfigParser has `setSerializationInclusion(JsonInclude.Include.NON_DEFAULT)`. So the field is omitted at serialization time when it's false. This breaks jackson's deduction.